### DISCUSSION TITLE
Ability to Override Primary Key Config for Implicit Join?

### DISCUSSION THREAD

**Discussion by brookeatgov** [prisma/prisma#25066] [Q&A]
### Question

We have several implicit join tables that don't have primary keys--they just have unique constraints on the composite key fields.

Debezium requires a primary key in order to track changes to tables, so we need to define the composite key as an actual primary key and not just a constraint.

We've tried creating a migration that specifically adds the key; however, prisma removes it on the next migration.

We don't want to change the schema to use explicit join models because we'd have to refactor every piece of code that uses this join, which is a lot of work and not low-risk.

We're hoping there's some decorator, etc that we could use to tell Prisma to leave our primary key in place.

Any ideas?

Thanks in advance!

### How to reproduce (optional)

```
DROP INDEX IF EXISTS "_groups_AB_unique";

ALTER TABLE _groups
    DROP CONSTRAINT IF EXISTS "_groups_pkey";

ALTER TABLE _groups
    ADD PRIMARY KEY ("A", "B");

### Expected behavior (optional)

Prisma should maintain the primary key we assigned to the table.

### Information about Prisma Schema, Client Queries and Environment (optional)

```
model Contact { 
    ...
    childContacts   Contact[]   @relation("groups")
    parentContacts  Contact[]   @relation("groups")
}

**Comment by ludralph** [comment]
Hi @brookeatgov 👋 

Thank you for raising this question. 

For implicit join tables, Prisma manages the primary keys internally and there is no decorator or directive to instruct Prisma to retain manually added primary keys in implicit join tables. You will need to use explicit many-to-many relations and manually create the join model with a primary key. This approach ensures that Prisma maintains and does not remove the primary key during migrations.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
